book/src/concepts/use_cases.md

```diff
@@ -0,0 +1,53 @@
+# Use Cases
+
+The following are examples of people who are well-suited to using a Portal Network client, like Trin.
+
+## Laptop wallet user
+
+A user has a laptop that frequently is turned off. When
+they want to transact, they can turn on Trin and connect their
+wallet to it.
+
+*Benefit*: Wallet use without reliance on third party wallet APIs.
+
+## Desktop wallet user
+
+A user has a desktop that usually on, but most of the disk is used for other things.
+When they want to transact, their wallet is already connected to their portal node.
+
+*Benefit*: Wallet use without reliance on third party wallet APIs. Contributes to
+network health without using entire disk.
+
+## Protocol experimentation
+
+A researcher looking to explore the Ethereum protocol, testing out
+specific aspects and perhaps making experimental changes to the protocol.
+
+*Benefit*: Spin up a node and play around quickly and with low cost.
+
+## Single board computer hobbyist
+
+A raspberry pi 3, or similarly-sized computer with could contribute
+to network health.
+
+Currently a raspberry pi 4 can run a full node, with consensus
+and execution clients, however this is a bit tight and requires a ~2TB SSD.
+
+*Benefit*: Learn about Ethereum, get node access and provide the
+network with additional robustness.
+
+## Mobile user
+
+Trin is not currently configured to run on mobile, however this is plausibly
+a viable and interesting use case for the future. A trin node could run as a
+background task with configurable limits on disk, CPU and bandwidth use.
+
+*Benefit*: Wallet use without reliance on third party wallet APIs. Contributes to
+network health.
+
+## Unsuitable users
+
+There are situations where Trin is estimated to not be a good node choice:
+- Very speedy historical state access. It's possible to retrieve old state, but don't expect sub-second contract reads on state as viewed from a historical block.
+- Building blocks locally, as a block producer. Random access to the full state and transaction pool is not supported at the speed needed to build competitive blocks.
+  - We remain hopeful that in the future, you could use an externally-generated block (like one provided by MEV-boost) so that you can act as a validater using a standard Consensus client, with Trin as the Execution client. This probably depends on a future where state witnesses are bundled with the block sent to you by the producer.
```

Comment:
```suggestion
  - Using  [Trin Execution](https://github.com/ethereum/trin/blob/master/trin-execution/README.md), an Execution client, which uses Trin to enable EIP-4444's well-maintaining access to Ethereum's full history for running a validator or a full node.
```

I am a bit confused what this is trying to convey. I don't think `Trin` is going to implement the engine API. This use case seems more like the grounds of `Trin Execution`. Even then `Trin Execution` would have the full head state so well someone could use `externally-generated block` it wouldn't be needed.

I don't understand the current paragraph because if `future where state witnesses are bundled with the block`, then the CL might as well just use an EVM library so I am not sure I see the value Trin would be providing here. I think the EL usecase is covered by Trin Execution and integrating this functionality into the Trin binary would be substracting from the value Trin provides and also would add unnecessary complexity.

Reply:
> I don't think `Trin` is going to implement the engine API. This use case seems more like the grounds of `Trin Execution`

From a user's perspective, I think we want `trin` to be the entry point that then launches whatever modules need to be launched to accomplish the goal that's being asked of it, including serving the engine API via block execution.

Since this is all future speculation, and framed as such in the book, I don't think we need to have perfect consensus here. We can update the book, as our understanding grows.

> I don't understand the current paragraph because if `future where state witnesses are bundled with the block`, then the CL might as well just use an EVM library

That seems a plausible direction for some CLs to go, for verifying blocks being sent from other proposers. But if it's your slot to propose, I doubt that all CL clients are going to get rid of the engine calls altogether. I don't think they want to get in the business of requesting the MEV-boost block and verifying it. (open to hearing from them, though!)

Depending on the direction of research, your local client may select some transactions to append onto prebuilt blocks. I don't think the CL wants to get into that either.

So I still think there's room for us to aim at serving the engine API for stakers.

Reply:
> > I don't think `Trin` is going to implement the engine API. This use case seems more like the grounds of `Trin Execution`
> 
> From a user's perspective, I think we want `trin` to be the entry point that then launches whatever modules need to be launched to accomplish the goal that's being asked of it, including serving the engine API via block execution.

I want to eventually spin `Trin Execution` outside of Trin and EF Portal and have it built and maintained by an independent team from Trin. Maybe changing `Trin Execution`'s name would be beneficial to differentiate the projects and prevent confusion. This has at least been my aspirations from the start of building `Trin Execution`. If `Trin Execution` was spun-out I would want to be the lead for the `Trin-Execution` team.

I think it would be best to focus `Trin` on being the best Portal client it can be. Well the idea of an everything app could be cool, often having "everything" dilutes the UX of a project. A jack of all trades is a master of none. It is hard to make something very good at something if it tries to do everything. I think Trin's win condition is being integrated into every app under the sun. Trin being written in Rust makes it a prime candidate to do that. Trin being a clean, extendible, performant implementation of Portal. Most people using Trin shouldn't know that they are using it. If Trin is the default Portal implementation people look to integrate, then Trin will spread faster than our small team can imagine.

I think we should discuss what the future of `Trin Execution` will be during Devcon.

Reply:
Cool yeah, let's chat in person. Lot's of interesting future plans to consider.